tools/cprnc/CMakeLists.txt

```diff
@@ -53,9 +48,6 @@ SET (CPRNC_SRCS
 ADD_EXECUTABLE(cprnc ${CPRNC_SRCS})
 ADD_DEPENDENCIES(cprnc COMPARE_VARS)
 
-TARGET_LINK_LIBRARIES(cprnc ${NETCDF_LIBRARIES})
-
-INSTALL(TARGETS cprnc
-        RUNTIME DESTINATION bin)
-
-execute_process(COMMAND ../configure --clean WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
+# SLIBS comes from Macros.cmake
+separate_arguments(SLIBS_LIST UNIX_COMMAND "${SLIBS}")
+TARGET_LINK_LIBRARIES(cprnc ${SLIBS_LIST})
```

Comment:
It looks like the assumption here is that the netcdf libraries are listed in SLIBS. This doesn't seem to be the case in general, at least for CESM machines. I think we need a more robust / general way of getting the netcdf libraries.

Reply:
How does CESM get netcdf libs?

Reply:
I think based on logic in the Makefile, at least for some machines. I can't comment on what the vision is for how it *should* be done, though.

Reply:
The problem is every system does it differently, and for a while even every version of netcdf did it differently - though that's long enough in the past that I'm not seeing the problem anymore.   On cheyenne for example when you load the netcdf module it adds the include path and the link steps to the mpi compiler wrappers so that you really don't need to explicitly do anything, on tacc systems they define TACC_NETCDF_PATH and on cray they define CRAY_NETCDF_PATH so then you need to translate that to what the model wants - usually that translation happens in config_compilers.xml - then there is cmake that wants a FindNetcdf.cmake file of which there are a lot out there and non of them are standard.   Oh if only it could be standard...

Reply:
What should we do with this one? I'm okay bringing this PR in as is, but opening an issue that this will need to be generalized. @jedwards4b @jgfouca @WesCoomber do you feel that's the best plan, or should this be fixed for this PR so that it's more likely to work across systems?

Reply:
We should make some effort to generalize this.

Reply:
I would like to get this into the ufs_release branch before a code freeze this Friday so I vote for merging and opening a new issue.

Reply:
I'm fine with that, though you may want to confirm that this will actually work on the machines you need it on for the ufs_release (it didn't work out of the box on my mac with the way I had things set up).

Reply:
So this is a cmake build - what about using this?
https://github.com/ESMCI/cime/blob/master/src/externals/pio2/cmake/FindNetCDF.cmake

Reply:
It seems like there must be something we can rely on to find netcdf. Any env variable or case setting?